### PR TITLE
External input

### PR DESCRIPTION
**Pull request by jan-janssen**
Access input of a scriptjob by:
```
from pyiron import ExternalInput
input_dict = ExternalInput()
```
this is the same as:
```
from pyiron import Notebook
input_dict = Notebook().get_custom_dict()
```
Besides the scriptjob input also JSON files as input are supported, just place a JSON file named "input.json" next to the notebook.

### CODE REVIEW

**Comment by jan-janssen** [comment]
The support for JSON files was added in https://github.com/pyiron/pyiron/pull/717

**Comment by samwaseda** [comment]
I don't really understand why these two different ways should coexist. Can you explain the advantage of the new method, and if it's convincing I'd rather put a deprecation warning for the old one.

**Comment by jan-janssen** [comment]
I find it more intuitive. If you want to we can deprecate the old function.

**Comment by samwaseda** [comment]
I find both of them not particularly intuitive, because they differ significantly from what we do in pyiron, i.e. we have to look it up anyway. What's for me intuitive would be something like

```
pr = Project('some_project')
input_dict = pr.load_input()
```

**Comment by jan-janssen** [comment]
That sounds good to me as well, I update the PR accordingly.

**Comment by jan-janssen** [comment]
@samwaseda Can you take another look?

File: pyiron/base/job/external.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    @staticmethod
+    def store_custom_output_dict(output_dict):
+        folder = Path(".").cwd().parts[-1]
+        hdf_file = Path(".").cwd().parents[1] / folder
+        hdf_file = str(hdf_file) + ".h5"
+        hdf = FileHDFio(hdf_file)
+        hdf[folder].create_group("output")
+        for k, v in output_dict.items():
+            hdf[folder + "/output"][k] = v
```
Comment (samwaseda) [inline pyiron/base/job/external.py:62]: Isn't it better to make the path the same as `input` or at least similar to it?

Comment (jan-janssen) [reply pyiron/base/job/external.py:62]: I guess that would be another pull request, so far I only copied your code ;-)

Comment (samwaseda) [reply pyiron/base/job/external.py:62]: I'm okay with having another pull request but apparently it was [your change](https://github.com/pyiron/pyiron/commit/45b9e31c51b95dcc13a7659ddcee59800ef3f2f8) 😎

**Comment by samwaseda** [comment]
I guess without Docstring it's sort of impossible for the user to figure out how to use this function.

**Comment by samwaseda** [comment]
And I guess it would be nice to give an error message or simply print a statement if the function is called when the input data is empty, because I'm pretty sure that one of the error sources will be that there's an error in communication between the notebooks/jobs and somehow the data gets lost.

**Comment by jan-janssen** [comment]
@samwaseda I added both the DocString and the Error message. If you agree, I would like to merge this PR now.

**Pull request merged**